.github/workflows/rust.yml

```diff
@@ -82,7 +86,10 @@ jobs:
       - uses: taiki-e/install-action@cargo-hack
       # No default features build
       - run: cargo build --target thumbv7em-none-eabi --release --no-default-features
-      - run: cargo hack build --target thumbv7em-none-eabi --release --each-feature --exclude-features default,std
+      - run: cargo hack build --target thumbv7em-none-eabi --release --each-feature --exclude-features default,std,legacy_compatibility
+      - env:
+          RUSTFLAGS: '-D warnings -A deprecated' # legacy_compatibility triggers a deprecation warning
```

Comment:
You can use `#[allow(deprecated)]` attributes instead of passing `-A deprecated` here, which will allow it to build without warnings

Reply:
Done. See latest 2 commits.